### DISCUSSION TITLE
Bun monorepo and type inference

### DISCUSSION THREAD

**Discussion by artecoop** [honojs/hono#1509] [Q&A] [answered]
Hello all,
I've created a bun monorepo (https://bun.sh/guides/install/workspaces) with a server and a client packages.

On the server package, I have an auth route:
```ts
import { Hono } from 'hono';
import { HTTPException } from 'hono/http-exception';

const routes = new Hono()
	.post('/login', async (c) => {
		try {
			return c.jsonT(...)
		} catch (e) {
			throw new HTTPException(401);
		}
	})
	.post('/logout', async (c) => {
			return c.body(null, 204);
	});

export default routes;
```

and the main entry point 
```ts
import { Hono } from 'hono';

import auth from '~routes/auth';

const app = new Hono();

const routes = app.route('/auth', auth);

export default {
	port: 6001,
	fetch: app.fetch
};

export type AppType = typeof routes;
```

Now, if I create inside this very same package a "client", as follow: 
```ts
import { hc } from 'hono/client';
import { AppType } from '~server';

const client = hc<AppType>('/');

const foo = await client.auth.login.$post({...})
```

typescript return the correct type for the route's endpoint
<img width="592" alt="Screenshot 2023-09-26 at 17 42 12" src="https://github.com/honojs/hono/assets/52246878/5e4d8df9-39f5-47b9-9289-02681e4fe9d4">

But if I move to another package on the same monorepo, in my case using vite and solidjs, type inference is lost
```ts
import { hc } from 'hono/client';

import { AppType } from '@ernesto/server/src/server';

const client = hc<AppType>('http://localhost:6001', {
	fetch(input, requestInit, _Env, _executionCtx) {
		return fetch(input, {
			...requestInit,
			credentials: 'include'
		});
	}
});

client.auth.???
```
<img width="618" alt="Screenshot 2023-09-26 at 17 44 25" src="https://github.com/honojs/hono/assets/52246878/11365a1c-683a-46ae-94d5-7e01002d3d0d">

Any idea?

**Comment by jaredLunde** [comment]
You're importing from the source directory `@ernesto/server/src/server` as opposed to the package directory `@ernesto/server` so any `"types": ""` property or import map in your `package.json` is not going to be recognized by TypeScript.

**Reply by artecoop** [reply]
Thanks for your reply!
I originally never mapped any types on `package.json`, and even take your suggestion and and adding `"types": "./src/server.ts"` on the `@ernesto/server` `package.json` did not change the final beavior

**Reply by jaredLunde** [reply]
Did you swap out your import `@ernesto/server/src/server` for `@ernesto/server` too?

**Reply by artecoop** [reply]
Absolutely. 

In the meantime I've messed up a little with the `tsconfig.json` as I'm thinking is the main culprit of all this situation, but  to no avail

**Comment by fredrik-sogaard** [comment, marked as answer]
We had a similar problem, and it was due to path aliases in tsconfig. See here: https://github.com/honojs/hono/issues/1723

**Reply by artecoop** [reply]
Frederik thanks for your response. Since then I've moved this project to another stack, but I'll keep it as answer for another time.

**Comment by ibash** [comment]
For future travelers, updating hono and typescript to the latest version fixed this.
